components/Account.php

```diff
@@ -286,23 +286,19 @@ public function onRegister()
                 $data['password_confirmation'] = post('password');
             }
 
-            $rules = (new UserModel)->rules;
-
-            if ($this->loginAttribute() !== UserSettings::LOGIN_USERNAME) {
-                unset($rules['username']);
-            }
```

Comment:
@bombozama this part is important to keep, you can change it to `unset($user->rules['username']);` and put it right before `$user->validate()`.

Reply:
Hmm, took a look at the beforeValidate, seems kind of unnecessarily complicated. @bennothommo @daftspunk could you take a look at https://github.com/rainlab/user-plugin/commit/971ce8cdd20ab2b9c1bbad0b19af195bbcca30ac#r36870182 and give me your thoughts?

Reply:
In my opinion, this shouldn't hold this PR to be processed. We should open a new PR for the issue you mention above.

Reply:
@mjauvin It does have a bearing on the functionality that @LukeTowers linked above, so we do have to pause the PR for it to be properly considered.